src/Microsoft.Framework.PackageManager/SourceControl/SourceBuilder.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Newtonsoft.Json;
+using NuGet;
+
+namespace Microsoft.Framework.PackageManager.SourceControl
+{
+    internal class SourceBuilder
+    {
+        private readonly Reports _reports;
+        private readonly IPackageBuilder _packageBuilder;
+        private readonly Runtime.Project _project;
+
+        public SourceBuilder(Runtime.Project project, IPackageBuilder packageBuilder, Reports buildReport)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+            if (buildReport == null)
+            {
+                throw new ArgumentNullException(nameof(buildReport));
+            }
+            if (packageBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(packageBuilder));
+            }
+
+            _project = project;
+            _packageBuilder = packageBuilder;
+            _reports = buildReport;
+        }
+
+        public bool Build(string outputPath)
+        {
+            if (_project.Repository == null)
+            {
+                // Nothing to do
+                return true;
+            }
+
+            _reports.Information.WriteLine("Embedding repository information...");
+
+            string repositoryType;
+            if (!_project.Repository.TryGetValue(Constants.RepoTypeKey, out repositoryType) ||
+                string.IsNullOrEmpty(repositoryType))
+            {
+                WriteError("The project file contains a repository section but the repository type is missing.");
```

Comment:
`repository section` -> `"repository" property`? I guess section is not a common term we use when talking about JSON